src/_pytest/fixtures.py

```diff
@@ -1126,7 +1127,7 @@ def pytest_fixture_setup(
             # Don't show the fixture as the skip location, as then the user
             # wouldn't know which test skipped.
             e._use_item_location = True
-        fixturedef.cached_result = (None, my_cache_key, e)
+        fixturedef.cached_result = (None, my_cache_key, (e, e.__traceback__))
```

Comment:
Would it be possible to just pass the dunder traceback into the method to avoid tracking the concrete one for reraise?

Reply:
I don't think so, the exception is saved on the `FixtureDef` and can be needed again at an arbitrary point in the future without a caller-callee relation to the time it is saved.